### PR TITLE
docs: add example of nested named routes

### LINKED ISSUE #1921: Document Named Views and Nested Routes
### Version
3.0.1

### Reproduction link
Full problem Example
[https://jsfiddle.net/oaxs8ek8/](https://jsfiddle.net/oaxs8ek8/)

### Steps to reproduce
I have created a Homepage with  2 Named Views called for example RouteA and RouteB. 
Now RouteA and RouteB have 2 other nested routes like so RouteA1 and RouteA2 - RouteB1 and RouteB2.

```
const router = new VueRouter({
  routes: [
    {
      path: '/',
      components: {
        RouteA : Bar,
        RouteB: Baz
      }
    }
  ]
})
```

this work very well for the first route but then i cant continue the routing chain for the 2 differente route since i cant only do this: 

```
const router = new VueRouter({
  routes: [
    {
      path: '/',
      components: {
        RouteA : Bar,
        RouteB: Baz
      },
      children: [
        {
          path: '' // Load RouteA1 on RouteA
          component: UserProfile
        },
        {
          path: '', // Load RouteB1 on RouteB
          component: UserPosts
        }
      ]
    }
    }
  ]
})
```

There is no way to specify the different route.

A solution could be enable this for example:

```
const router = new VueRouter({
  routes: [
    {
      path: '/',
      components: {
        RouteA : {
                     path: ''
                     component: Bar,
                     children:{
                        ....
                     }
              },
        RouteB:  {
                     path: ''
                     component: Baz,
                     children:{
                        ....
                     }
      }
    }
  ]
})
```

### What is expected?
The router should allow nested route with named view router

### What is actually happening?
it's not routeing correctly

### PR DESCRIPTION
**Pull request by posva**
Closes #1921

### CODE REVIEW

File: docs/en/essentials/named-views.md [hunk 1]
```diff
@@ -27,3 +27,58 @@ const router = new VueRouter({
...
+<!-- UserSettings.vue -->
+<div>
+  <h1>User Settings</h1>
+  <NavBar/>
+  <router-view/>
+  <router-view name="helper"/>
+</div>
+```
+
+_The nested view components are omitted here but you can find the full example at the end in the jsfiddle_
```
Comment (Akryum) [inline docs/en/essentials/named-views.md]: It's unfortunate that the jsfiddle link is at the end. Reader may ask "what jsfiddle?" here.
Maybe add the link here too?

File: docs/en/essentials/named-views.md [hunk 2]
```diff
@@ -27,3 +27,58 @@ const router = new VueRouter({
...
 
 A working demo of this example can be found [here](https://jsfiddle.net/posva/6du90epg/).
+
+## Nested Named Views
+
+While being quite complex, it is possible to combine named views with nested view. When doing so, you will also need to name nested `router-view` components used. Let's take a Settings panel example:
+
+```
+/settings/emails                                       /settings/profile
++-----------------------------------+                  +------------------------------+
```
Comment (znck) [inline docs/en/essentials/named-views.md]: We should put some image instead.

File: docs/en/essentials/named-views.md [hunk 3]
```diff
@@ -27,3 +27,58 @@ const router = new VueRouter({
 ```
 
 A working demo of this example can be found [here](https://jsfiddle.net/posva/6du90epg/).
+
+## Nested Named Views
+
+While being quite complex, it is possible to combine named views with nested view. When doing so, you will also need to name nested `router-view` components used. Let's take a Settings panel example:
```
Comment (znck) [inline docs/en/essentials/named-views.md]: Re-wording.
---
It is possible to create complex layouts using named views with nested views. For example, let us see following example; a settings panel.

File: docs/en/essentials/named-views.md [hunk 4]
```diff
@@ -27,3 +27,58 @@ const router = new VueRouter({
...
+| |     |                         | |                  | |     | UserProfilePreview | |
+| +-----+-------------------------+ |                  | +-----+--------------------+ |
++-----------------------------------+                  +------------------------------+
+```
+
+- `Nav` is just a regular compnonent
+- `UserSettings` is the view comopnent
+- `UserEmailsSubscriptions`, `UserProfile`, `UserProfilePreview` are nested view components
+
+**Note**: _Let's forget about how the HTML/CSS should look like to represent such layout and focus on the components used_
```
Comment (znck) [inline docs/en/essentials/named-views.md:51]: Re-wording.
---
The `<template>` section for `UserSettings` component in above layout would be something like this:

File: docs/en/essentials/named-views.md [hunk 5]
```diff
@@ -27,3 +27,58 @@ const router = new VueRouter({
...
+<!-- UserSettings.vue -->
+<div>
+  <h1>User Settings</h1>
+  <NavBar/>
+  <router-view/>
+  <router-view name="helper"/>
+</div>
+```
+
+_The nested view components are omitted here but you can find the full example [here](https://jsfiddle.net/posva/22wgksa3/)_
```
Comment (znck) [inline docs/en/essentials/named-views.md]: Re-wording.
---
You can find the complete source code for above example [here](https://jsfiddle.net/posva/22wgksa3/).

File: docs/en/essentials/named-views.md (hunk 2 above)
Comment (posva) [reply docs/en/essentials/named-views.md]: Let's keep this for the moment. An Image would be indeed better

**Comment by posva** [comment]
@znck Thanks! I updated the relevant parts the best I could

File: docs/en/essentials/named-views.md (hunk 4 above)
Comment (posva) [reply docs/en/essentials/named-views.md:51]: I added this just below

**Review by znck — APPROVED** [review]

**Pull request merged**